### ISSUE TITLE
switching to collab access from public fails to correctly switch the sasurl.  

### ISSUE THREAD

**Issue by havok2063** [sdss/marvin#575]
**Describe the bug**
At least in the `vetround2 branch, starting with a `public` access mode and switching to `collab` fails to switch the sasurl needed. Then when switching to an MPL and trying to access content, things fail.  This may be related to #565.   

**To Reproduce**
1. Set `check_access` to False in the `marvin.yml`. 
2. Load marvin.  Default will be DR15. 
3. `config.access = "collab"`
4. `config.setRelease = "MPL-7"`
5. `from marvin.tools import Maps`
6. `m = Maps('8937-1901')  `

Produces the following traceback
```
/Users/Brian/anaconda2/envs/marvindev/lib/python3.5/site-packages/marvin/tools/mixins/mma.py:139: MarvinUserWarning: local mode failed. Trying remote now.
  warnings.warn('local mode failed. Trying remote now.', MarvinUserWarning)
release MPL-4
making remote call to /marvin/api/query/getallparams/ MPL-4
/Users/Brian/anaconda2/envs/marvindev/lib/python3.5/site-packages/marvin/utils/datamodel/query/base.py:118: MarvinUserWarning: Could not remotely retrieve full set of parameters. byte indices must be integers or slices, not str
  warnings.warn('Could not remotely retrieve full set of parameters. {0}'.format(e), MarvinUserWarning)
release MPL-5
making remote call to /marvin/api/query/getallparams/ MPL-5
release MPL-6
making remote call to /marvin/api/query/getallparams/ MPL-6
release MPL-7
making remote call to /marvin/api/query/getallparams/ MPL-7
release DR15
making remote call to /marvin/api/query/getallparams/ DR15
/Users/Brian/anaconda2/envs/marvindev/lib/python3.5/site-packages/marvin/utils/datamodel/query/base.py:118: MarvinUserWarning: Could not remotely retrieve full set of parameters. a bytes-like object is required, not 'str'
  warnings.warn('Could not remotely retrieve full set of parameters. {0}'.format(e), MarvinUserWarning)
/Users/Brian/anaconda2/envs/marvindev/lib/python3.5/importlib/_bootstrap.py:222: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  return f(*args, **kwds)
---------------------------------------------------------------------------
HTTPError                                 Traceback (most recent call last)
/Users/Brian/anaconda2/envs/marvindev/lib/python3.5/site-packages/marvin/extern/marvin_brain/python/brain/api/api.py in _checkResponse(self, response)
    197         try:
--> 198             isbad = response.raise_for_status()
    199         except requests.HTTPError as http:

/Users/Brian/anaconda2/envs/marvindev/lib/python3.5/site-packages/requests/models.py in raise_for_status(self)
    938         if http_error_msg:
--> 939             raise HTTPError(http_error_msg, response=self)
    940

HTTPError: 401 Client Error: Unauthorized for url: https://lore.sdss.utah.edu/marvin/api/maps/8937-1901/HYB10/GAU-MILESHC/

During handling of the above exception, another exception occurred:

TypeError                                 Traceback (most recent call last)
/Users/Brian/anaconda2/envs/marvindev/lib/python3.5/site-packages/marvin/tools/maps.py in _load_maps_from_api(self)
    369         try:
--> 370             response = self._toolInteraction(url_full)
    371         except Exception as ee:

/Users/Brian/anaconda2/envs/marvindev/lib/python3.5/site-packages/marvin/tools/core.py in _toolInteraction(self, url, params)
    131         params = params or {'release': self._release}
--> 132         return marvin.api.api.Interaction(url, params=params)
    133

/Users/Brian/anaconda2/envs/marvindev/lib/python3.5/site-packages/marvin/extern/marvin_brain/python/brain/api/api.py in __init__(self, route, params, request_type, auth, timeout, headers, stream, datastream, send)
     53         if self.url and send:
---> 54             self._sendRequest(request_type)
     55         elif not self.url and send:

/Users/Brian/anaconda2/envs/marvindev/lib/python3.5/site-packages/marvin/extern/marvin_brain/python/brain/api/api.py in _sendRequest(self, request_type)
    289             # Check the response if it's good
--> 290             self._checkResponse(self._response)
    291

/Users/Brian/anaconda2/envs/marvindev/lib/python3.5/site-packages/marvin/extern/marvin_brain/python/brain/api/api.py in _checkResponse(self, response)
    212                 elif self.authtype == 'token':
--> 213                     msg = '{0}. Please check your token or login again for a fresh one.'.format(json_data['msg'])
    214                 else:

TypeError: byte indices must be integers or slices, not str

During handling of the above exception, another exception occurred:

MarvinError                               Traceback (most recent call last)
<ipython-input-14-c4b5575a11ac> in <module>()
----> 1 m = Maps('8937-1901')

/Users/Brian/anaconda2/envs/marvindev/lib/python3.5/site-packages/marvin/tools/maps.py in __init__(self, input, filename, mangaid, plateifu, mode, data, release, drpall, download, nsa_source, bintype, template, template_kin)
    109             self._load_maps_from_db(data=self.data)
    110         elif self.data_origin == 'api':
--> 111             self._load_maps_from_api()
    112         else:
    113             raise marvin.core.exceptions.MarvinError(

/Users/Brian/anaconda2/envs/marvindev/lib/python3.5/site-packages/marvin/tools/maps.py in _load_maps_from_api(self)
    371         except Exception as ee:
    372             raise marvin.core.exceptions.MarvinError(
--> 373                 'found a problem when checking if remote maps exists: {0}'.format(str(ee)))
    374
    375         data = response.getData()

MarvinError: found a problem when checking if remote maps exists: byte indices must be integers or slices, not str.

In [15]: m = Maps('8937-1901')
/Users/Brian/anaconda2/envs/marvindev/lib/python3.5/site-packages/marvin/tools/mixins/mma.py:139: MarvinUserWarning: local mode failed. Trying remote now.
  warnings.warn('local mode failed. Trying remote now.', MarvinUserWarning)
---------------------------------------------------------------------------
HTTPError                                 Traceback (most recent call last)
/Users/Brian/anaconda2/envs/marvindev/lib/python3.5/site-packages/marvin/extern/marvin_brain/python/brain/api/api.py in _checkResponse(self, response)
    197         try:
--> 198             isbad = response.raise_for_status()
    199         except requests.HTTPError as http:

/Users/Brian/anaconda2/envs/marvindev/lib/python3.5/site-packages/requests/models.py in raise_for_status(self)
    938         if http_error_msg:
--> 939             raise HTTPError(http_error_msg, response=self)
    940

HTTPError: 401 Client Error: Unauthorized for url: https://lore.sdss.utah.edu/marvin/api/maps/8937-1901/HYB10/GAU-MILESHC/

During handling of the above exception, another exception occurred:

TypeError                                 Traceback (most recent call last)
/Users/Brian/anaconda2/envs/marvindev/lib/python3.5/site-packages/marvin/tools/maps.py in _load_maps_from_api(self)
    369         try:
--> 370             response = self._toolInteraction(url_full)
    371         except Exception as ee:

/Users/Brian/anaconda2/envs/marvindev/lib/python3.5/site-packages/marvin/tools/core.py in _toolInteraction(self, url, params)
    131         params = params or {'release': self._release}
--> 132         return marvin.api.api.Interaction(url, params=params)
    133

/Users/Brian/anaconda2/envs/marvindev/lib/python3.5/site-packages/marvin/extern/marvin_brain/python/brain/api/api.py in __init__(self, route, params, request_type, auth, timeout, headers, stream, datastream, send)
     53         if self.url and send:
---> 54             self._sendRequest(request_type)
     55         elif not self.url and send:

/Users/Brian/anaconda2/envs/marvindev/lib/python3.5/site-packages/marvin/extern/marvin_brain/python/brain/api/api.py in _sendRequest(self, request_type)
    289             # Check the response if it's good
--> 290             self._checkResponse(self._response)
    291

/Users/Brian/anaconda2/envs/marvindev/lib/python3.5/site-packages/marvin/extern/marvin_brain/python/brain/api/api.py in _checkResponse(self, response)
    212                 elif self.authtype == 'token':
--> 213                     msg = '{0}. Please check your token or login again for a fresh one.'.format(json_data['msg'])
    214                 else:

TypeError: byte indices must be integers or slices, not str

During handling of the above exception, another exception occurred:

MarvinError                               Traceback (most recent call last)
<ipython-input-15-c4b5575a11ac> in <module>()
----> 1 m = Maps('8937-1901')

/Users/Brian/anaconda2/envs/marvindev/lib/python3.5/site-packages/marvin/tools/maps.py in __init__(self, input, filename, mangaid, plateifu, mode, data, release, drpall, download, nsa_source, bintype, template, template_kin)
    109             self._load_maps_from_db(data=self.data)
    110         elif self.data_origin == 'api':
--> 111             self._load_maps_from_api()
    112         else:
    113             raise marvin.core.exceptions.MarvinError(

/Users/Brian/anaconda2/envs/marvindev/lib/python3.5/site-packages/marvin/tools/maps.py in _load_maps_from_api(self)
    371         except Exception as ee:
    372             raise marvin.core.exceptions.MarvinError(
--> 373                 'found a problem when checking if remote maps exists: {0}'.format(str(ee)))
    374
    375         data = response.getData()

MarvinError: found a problem when checking if remote maps exists: byte indices must be integers or slices, not str.
```
** Additional Context **
Looking through the logs when this occurred, I can't find any trace of an original error.

**Comment by havok2063** [comment]
I think this issue was unrelated to the `sasurl` not correctly changing.  The `sasurl` does not change when switching access modes.  It changes when switching releases.  Using the code above to reproduce the following error:

When trying to load the maps, the expected result is an error indicating you need an access token.  The following error when attempting to use `config.login` was the underlying bug.  

```
from marvin import config, marvindb
INFO: No release version set. Setting default to DR15
config.access = 'collab'
config.setRelease('MPL-7')
config.sasurl
'https://lore.sdss.utah.edu/'

from marvin.tools import Maps
maps = Maps('8937-1901')

---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
~/Work/github_projects/Marvin/python/marvin/tools/maps.py in _load_maps_from_api(self)
    370         try:
--> 371             response = self._toolInteraction(url_full)
    372         except Exception as ee:

~/Work/github_projects/Marvin/python/marvin/tools/core.py in _toolInteraction(self, url, params)
    131         params = params or {'release': self._release}
--> 132         return marvin.api.api.Interaction(url, params=params)
    133

~/Work/github_projects/Marvin/python/marvin/extern/marvin_brain/python/brain/api/api.py in __init__(self, route, params, request_type, auth, timeout, headers, stream, datastream, send, base)
     54         if self.url and send:
---> 55             self._sendRequest(request_type)
     56         elif not self.url and send:

~/Work/github_projects/Marvin/python/marvin/extern/marvin_brain/python/brain/api/api.py in _sendRequest(self, request_type)
    276                 self._response = self.session.post(self.url, data=self.params, timeout=self.timeout,
--> 277                                                    headers=self.headers, stream=self.stream)
    278         except requests.Timeout as rt:

~/anaconda2/envs/marvin-py3/lib/python3.5/site-packages/requests/sessions.py in post(self, url, data, json, **kwargs)
    521
--> 522         return self.request('POST', url, data=data, json=json, **kwargs)
    523

~/anaconda2/envs/marvin-py3/lib/python3.5/site-packages/requests/sessions.py in request(self, method, url, params, data, headers, cookies, files, auth, timeout, allow_redirects, proxies, hooks, stream, verify, cert, json)
    460         )
--> 461         prep = self.prepare_request(req)
    462

~/anaconda2/envs/marvin-py3/lib/python3.5/site-packages/requests/sessions.py in prepare_request(self, request)
    393             cookies=merged_cookies,
--> 394             hooks=merge_hooks(request.hooks, self.hooks),
    395         )

~/anaconda2/envs/marvin-py3/lib/python3.5/site-packages/requests/models.py in prepare(self, method, url, headers, files, data, params, auth, cookies, hooks, json)
    297         self.prepare_body(data, files, json)
--> 298         self.prepare_auth(auth, url)
    299

~/anaconda2/envs/marvin-py3/lib/python3.5/site-packages/requests/models.py in prepare_auth(self, auth, url)
    499             # Allow auth to make its changes.
--> 500             r = auth(self)
    501

~/Work/github_projects/Marvin/python/marvin/extern/marvin_brain/python/brain/api/api.py in __call__(self, r)
    349         if self.authtype == 'token':
--> 350             assert bconfig.token is not None, 'You must have a valid token set to use the API.  Please login.'
    351             r.headers['Authorization'] = 'Bearer {0}'.format(bconfig.token)

AssertionError: You must have a valid token set to use the API.  Please login.

During handling of the above exception, another exception occurred:

MarvinError                               Traceback (most recent call last)
<ipython-input-6-69029465a80c> in <module>()
----> 1 maps = Maps('8937-1901')

~/Work/github_projects/Marvin/python/marvin/tools/maps.py in __init__(self, input, filename, mangaid, plateifu, mode, data, release, drpall, download, nsa_source, bintype, template, template_kin)
    110             self._load_maps_from_db(data=self.data)
    111         elif self.data_origin == 'api':
--> 112             self._load_maps_from_api()
    113         else:
    114             raise marvin.core.exceptions.MarvinError(

~/Work/github_projects/Marvin/python/marvin/tools/maps.py in _load_maps_from_api(self)
    372         except Exception as ee:
    373             raise marvin.core.exceptions.MarvinError(
--> 374                 'found a problem when checking if remote maps exists: {0}'.format(str(ee)))
    375
    376         data = response.getData()

MarvinError: found a problem when checking if remote maps exists: You must have a valid token set to use the API.  Please login.
```

config.login()
```
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
~/Work/github_projects/Marvin/python/marvin/__init__.py in login(self, refresh)
    710             try:
--> 711                 resp = Interaction(url, params=data, auth='netrc')
    712             except Exception as e:

~/Work/github_projects/Marvin/python/marvin/extern/marvin_brain/python/brain/api/api.py in __init__(self, route, params, request_type, auth, timeout, headers, stream, datastream, send, base)
     54         if self.url and send:
---> 55             self._sendRequest(request_type)
     56         elif not self.url and send:

~/Work/github_projects/Marvin/python/marvin/extern/marvin_brain/python/brain/api/api.py in _sendRequest(self, request_type)
    276                 self._response = self.session.post(self.url, data=self.params, timeout=self.timeout,
--> 277                                                    headers=self.headers, stream=self.stream)
    278         except requests.Timeout as rt:

~/anaconda2/envs/marvin-py3/lib/python3.5/site-packages/requests/sessions.py in post(self, url, data, json, **kwargs)
    521
--> 522         return self.request('POST', url, data=data, json=json, **kwargs)
    523

~/anaconda2/envs/marvin-py3/lib/python3.5/site-packages/requests/sessions.py in request(self, method, url, params, data, headers, cookies, files, auth, timeout, allow_redirects, proxies, hooks, stream, verify, cert, json)
    460         )
--> 461         prep = self.prepare_request(req)
    462

~/anaconda2/envs/marvin-py3/lib/python3.5/site-packages/requests/sessions.py in prepare_request(self, request)
    393             cookies=merged_cookies,
--> 394             hooks=merge_hooks(request.hooks, self.hooks),
    395         )

~/anaconda2/envs/marvin-py3/lib/python3.5/site-packages/requests/models.py in prepare(self, method, url, headers, files, data, params, auth, cookies, hooks, json)
    297         self.prepare_body(data, files, json)
--> 298         self.prepare_auth(auth, url)
    299

~/anaconda2/envs/marvin-py3/lib/python3.5/site-packages/requests/models.py in prepare_auth(self, auth, url)
    499             # Allow auth to make its changes.
--> 500             r = auth(self)
    501

~/Work/github_projects/Marvin/python/marvin/extern/marvin_brain/python/brain/api/api.py in __call__(self, r)
    349         if self.authtype == 'token':
--> 350             assert bconfig.token is not None, 'You must have a valid token set to use the API.  Please login.'
    351             r.headers['Authorization'] = 'Bearer {0}'.format(bconfig.token)

AssertionError: You must have a valid token set to use the API.  Please login.

During handling of the above exception, another exception occurred:

MarvinError                               Traceback (most recent call last)
<ipython-input-7-e2eb019c43e1> in <module>()
----> 1 config.login()

~/Work/github_projects/Marvin/python/marvin/__init__.py in login(self, refresh)
    711                 resp = Interaction(url, params=data, auth='netrc')
    712             except Exception as e:
--> 713                 raise MarvinError('Error getting login token. {0}'.format(e))
    714             else:
    715                 self.token = resp.results['access_token']

MarvinError: Error getting login token. You must have a valid token set to use the API.  Please login.
```
Logging in uses the `netrc` authentication method.  This method was not being propagating correctly all the way into the `BrainAuth` class used by `BrainInteraction`.   This has been fixed in the `brain`.  

After the recent fix, the following code works after following the inital steps up to the first `Maps` access error.   I think this is fixed and also fixes #565.   I'll mark this as needs reprodution rather than closing.    

```
config.login()
maps = Maps('8937-1901')
maps
<Marvin Maps (plateifu='8937-1901', mode='remote', data_origin='api', bintype='HYB10', template='GAU-MILESHC')>
```

**Comment by bretthandrews** [comment]
Works as intended on non-dev machine.
